drivers/i2c/Kconfig.stm32

```diff
@@ -59,7 +60,7 @@ config I2C_STM32_V2_DMA
 	bool "DMA support"
 	depends on I2C_STM32_V2
 	select DMA
-	select CACHE_MANAGEMENT if CPU_HAS_DCACHE
+	select CACHE_MANAGEMENT if DCACHE
```

Comment:
possibly outside of the remit of this PR but does this suggest where #ifdef CONFIG_DCACHE is used in the i2c driver should change to #ifdef CACHE_MANAGEMENT instead?

Reply:
Ah ok, thanks for the link!